### PR TITLE
Remove cogateway deactivation logic

### PR DESCRIPTION
**Pull request by sarvesh-ost**
This should be merged after #512  

Removed deactivation logic from `GatewayBase` and now only gateway and co-gateway can be deactivated. Also, the stake can only be called if the gateway is activated.

### CODE REVIEW

**Comment by schemar** [comment]
@sarvesh-ost can you please merge develop and resolve the conflicts before we review it?

**Comment by sarvesh-ost** [comment]
> @sarvesh-ost can you please merge develop and resolve the conflicts before we review it?

It's now available for review.

File: contracts/gateway/EIP20Gateway.sol [hunk 1]
```diff
@@ -197,6 +200,17 @@ contract EIP20Gateway is GatewayBase {
     /** Maps messageHash to the Unstake object. */
     mapping(bytes32 /*messageHash*/ => Unstake) unstakes;
 
+    /* modifiers */
+
+    /** checks that contract is activated */
```
Comment (schemar) [inline contracts/gateway/EIP20Gateway.sol]:
```suggestion
    /** Checks that contract is active. */
```

File: test/gateway/eip20_gateway/stake.js [hunk 2]
```diff
@@ -317,5 +318,24 @@ contract('EIP20Gateway.stake() ', function (accounts) {
...
+            mockToken.address,
+            baseToken.address,
+            accounts[1], //core address
+            bountyAmount,
+            accounts[2] // organisation address
+        );
+
+        await _setup(accounts, gateway);
+        await _prepareData();
+        await _stake(utils.ResultType.FAIL);
```
Comment (schemar) [inline test/gateway/eip20_gateway/stake.js]: Where is the gateway activated in the other test cases? 😕

Comment (sarvesh-ost) [reply test/gateway/eip20_gateway/stake.js]: In the mock contract, it has been set to` true` on construction.
[Link](https://github.com/sarvesh-ost/mosaic-contracts/blob/daa109f14639a7677867e206fab85c0b3ee6a3e6/contracts/test/TestEIP20Gateway.sol#L58)

**Pull request merged**